pkg/storage/factory.go

```diff
@@ -198,11 +232,20 @@ func NewIndexClient(name string, cfg Config, schemaCfg config.SchemaConfig, limi
 
 // NewChunkClient makes a new chunk.Client of the desired types.
 func NewChunkClient(name string, cfg Config, schemaCfg config.SchemaConfig, clientMetrics ClientMetrics, registerer prometheus.Registerer) (client.Client, error) {
-	switch name {
+	var (
+		storeType = name
+	)
+
+	// extact storeType for named stores
+	if words := strings.SplitN(name, ".", 2); len(words) == 2 {
```

Comment:
Rather than requiring we use `<type>.<name>` as a convention in the configuration, what if we allowed specifying only the store's name and resolved the type from that. As a bonus, this would also work with stores that used the reserved `.` characters, such as `name: "my.s3.compatible.store"`. I think this would be easier to use and understand (I had to check test cases to realize the convention of putting the `.` in the store name to look it up). Note: if we made this change, all the individual types would be reserved store names (for example, you couldn't redefine another store with name `aws`).

Reply:
I was also a bit worried that this convention might not be the most intuitive.
Just to make sure I understand it correctly: we want to drop the type prefix when referring to the named store, but the way we are configuring names stores can remain the same right?